ament_cmake_core/cmake/symlink_install/ament_cmake_symlink_install_targets.cmake

```diff
@@ -89,8 +85,19 @@ function(ament_cmake_symlink_install_targets)
     string(REPLACE "\"LIBRARY\" \"DESTINATION\"" "\"LIBRARY_DESTINATION\"" argn_quoted "${argn_quoted}")
     string(REPLACE "\"RUNTIME\" \"DESTINATION\"" "\"RUNTIME_DESTINATION\"" argn_quoted "${argn_quoted}")
 
+    # generate unique files
+    set(generated_file
+      "${CMAKE_CURRENT_BINARY_DIR}/ament_cmake_symlink_install_targets_${__AMENT_CMAKE_SYMLINK_INSTALL_TARGETS_INDEX}.cmake")
+    math(EXPR __AMENT_CMAKE_SYMLINK_INSTALL_TARGETS_INDEX
+      "${__AMENT_CMAKE_SYMLINK_INSTALL_TARGETS_INDEX} + 1")
+    set(__AMENT_CMAKE_SYMLINK_INSTALL_TARGETS_INDEX "${__AMENT_CMAKE_SYMLINK_INSTALL_TARGETS_INDEX}"
+      CACHE INTERNAL "Index for unique symlink install targets")
```

Comment:
Why is this necessary? What's the purpose of creating unique files with the install targets in them?